docs/extensibility/snippets/MailDevResource/MailKit.Client/MailKitClientServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,146 @@
+    /// read from the configuration.
+    /// </param>
+    public static void AddKeyedMailKitClient(
+        this IHostApplicationBuilder builder,
+        string name,
+        Action<MailKitClientSettings>? configureSettings = null)
+    {
+        ArgumentNullException.ThrowIfNull(name);
+
+        AddMailKitClient(
+            builder,
+            $"{DefaultConfigSectionName}:{name}",
+            configureSettings,
+            connectionName: name,
+            serviceKey: name);
+    }
+
+    private static void AddMailKitClient(
+        this IHostApplicationBuilder builder,
+        string configurationSectionName,
+        Action<MailKitClientSettings>? configureSettings,
+        string connectionName,
+        object? serviceKey)
+    {
+        ArgumentNullException.ThrowIfNull(builder);
+        var settings = new MailKitClientSettings();
+
+        builder.Configuration
+               .GetSection(configurationSectionName)
+               .Bind(settings);
+
+        if (builder.Configuration.GetConnectionString(connectionName) is string connectionString)
+        {
+            settings.ConnectionString = connectionString;
+        }
+
+        configureSettings?.Invoke(settings);
+
+        if (serviceKey is null)
+        {
+            builder.Services.AddScoped(CreateMailKitClientFactory);
+        }
+        else
+        {
+            builder.Services.AddKeyedScoped(serviceKey, (sp, _) => CreateMailKitClientFactory(sp));
+        }
+
+        MailKitClientFactory CreateMailKitClientFactory(IServiceProvider _)
+        {
+            var connectionString = settings.ConnectionString;
+
+            if (string.IsNullOrWhiteSpace(connectionString))
+            {
+                throw new InvalidOperationException($"""
+                    ConnectionString is missing.
+                    It should be provided in 'ConnectionStrings:{connectionName}'
+                    or under 'ConnectionString' key in '{DefaultConfigSectionName}'
+                    configuration section.
+                    """);
+            }
+
+            if (Uri.TryCreate(connectionString, UriKind.Absolute, out var smtpUri) is false ||
+                smtpUri is null)
+            {
+                throw new InvalidOperationException($"""
+                    The 'ConnectionStrings:{connectionName}' (or 'ConnectionString' key in
+                    '{DefaultConfigSectionName}') isn't a valid URI format.
+                    """);
+            }
+
+            return new MailKitClientFactory(smtpUri, settings.Credentials);
+        }
+
+        if (settings.DisableHealthChecks is false)
+        {
+            builder.Services.AddHealthChecks()
+                .AddCheck<MailKitHealthCheck>(
+                    name: serviceKey is null ? "MailKit" : $"MailKit_{connectionName}",
+                    failureStatus: default,
+                    tags: [ "live" ]);
```

Comment:
We don't add this tag to our health checks. Full health checks don't get run on `/alive` checks.